standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/events/CommitCompactionEvent.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore.events;
+
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.hive.metastore.IHMSHandler;
+import org.apache.hadoop.hive.metastore.api.CompactionType;
+import org.apache.hadoop.hive.metastore.api.TxnType;
+import org.apache.hadoop.hive.metastore.txn.CompactionInfo;
+
+/**
+ * CommitCompactionEvent
+ * Event generated for compaction commit operation
+ */
+@InterfaceAudience.Public
+@InterfaceStability.Stable
+public class CommitCompactionEvent extends ListenerEvent {
+
+  private final Long txnId;
+  private final Long compactionId;
+  private final CompactionType type;
+  private final String dbname;
```

Comment:
I haven't seen that other events contained db information. Is it required here?

Reply:
The filemetadata cache will have to invalidate the cache of a given partition based on the event payload. It most likely  will need to use the dbName as part of the key, to identify the right table. When it matters the event contains this info (see AllocWriteIdEvent)